src/record.py

```diff
@@ -20,6 +20,9 @@ def __init__(self, filename, card=None):
         else:
             self.card = card
             self.parser = get_parser(card)
+        if df is not None:
+            self._records = df.copy(deep=True)
```

Comment:
I'm not sure if this should be `self._records = df` or if we need the deepcopy. The way it's used I think just = df is fine, but I'm not sure what the best practice here is.